Team9Project/src/com/indragie/cmput301as1/BaseTabFragment.java

```diff
@@ -0,0 +1,292 @@
+package com.indragie.cmput301as1;
+ 
+import java.util.Comparator;
+import java.util.List;
+
+import android.app.AlertDialog;
+import android.app.ListFragment;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.ActionMode;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemLongClickListener;
+import android.widget.ListView;
+
+ 
+
+public class BaseTabFragment extends ListFragment implements TypedObserver<CollectionMutation<ExpenseClaim>> {
+ 
+    protected static final String EXPENSE_CLAIM_FILENAME = "claims";
+    protected static final String EXPENSE_APPROVAL_FILENAME = "approval";
+    private static final String PREFERENCE = "PREFERENCE";
+	private static final int ADD_EXPENSE_CLAIM_REQUEST = 1;
+	private static final int EDIT_EXPENSE_CLAIM_REQUEST = 2;
+	private static final int SORT_EXPENSE_CLAIM_REQUEST = 3;
+    
+    protected ListModel<ExpenseClaim> listModel;
+    
+    ListView myButton;
+    /**
+	 * Index of a item that is long pressed.
+	 */
+	private int longPressedItemIndex;
+	
+	/**
+	 * Active user.
+	 */
+	private User user;
+
+	/** An array of items to display in ArrayList */
+    
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    	
+    	setUserFromPreferences();
+    	setHasOptionsMenu(true);
+        
+    	
+    	load();
+		
+    	/** Creating array adapter to set data in listview */
+        ExpenseClaimArrayAdapter adapter = new ExpenseClaimArrayAdapter(getActivity(), listModel.getItems());
+ 
+        /** Setting the array adapter to the listview */
+        setListAdapter(adapter);
+        
+        
+		
+ 
+        return super.onCreateView(inflater, container, savedInstanceState);
+    }
+    
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState){
+    	super.onActivityCreated(savedInstanceState);
+    	
+    	
+    	final ActionMode.Callback longClickCallback = new ActionMode.Callback() {
```

Comment:
This should use `LongClickDeleteListener` instead, it eliminates a lot of this boilerplate code